app/src/main/java/com/example/erabook/fragments/discoverGoogle/DiscoverDetailsFragment.kt

```diff
@@ -90,42 +91,47 @@ class DiscoverDetailsFragment : Fragment() {
                 "ALL_PAGES" -> {
                     bookItem.accessInfo.pdf?.let { pdf ->
                         if (pdf.isAvailable == true) {
+                            isDownloadableButton.setBackgroundResource(R.drawable.download)
                             isDownloadableButton.setOnClickListener {
                                 openLinkBookDownload(
                                     pdf.downloadLink,
                                     requireContext()
                                 )
                             }
-                        } else {
-                            isDownloadableButton.setImageResource(R.drawable.no_download)
+                        } else if (pdf.isAvailable == false) {
+                            isDownloadableButton.setBackgroundResource(R.drawable.no_download)
                         }
                     }
+                }
 
-                    authenticationViewModel.userLiveData.observe(viewLifecycleOwner) { user ->
-                        favoriteBook.setOnClickListener {
-                            sharedViewModel.saveBookToDB(bookItem, user?.email.toString())
-                            sharedViewModel.isSaved.observe(viewLifecycleOwner) { value ->
-                                if (value) {
-                                    favoriteBook.setImageResource(R.drawable.favorite)
-                                    requireActivity().showToast(R.string.favorite_added)
-                                } else {
-                                    requireActivity().showToast(R.string.update_message_error)
-                                }
-                            }
-                        }
-                    }
-                    shareBook.setOnClickListener {
-                        bookItem.volumeInfo?.let {
-                            requireActivity().startBookDetailsIntent(
-                                it.title,
-                                it.authors[0],
-                                it.pageCount.toString(),
-                                it.publishedDate
-                            )
+                else -> {
+                    isDownloadableButton.setBackgroundResource(R.drawable.no_download)
```

Comment:
Looks like 'isDownloadButton' is a button on the UI right? If so why it has the prefix "is", doesn't make sense to me.
That prefix is useful when dealing with booleans, otherwise you can safely rename it to downloadButton.